graph/number_of_paths.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file
+ * @brief Algorithm to count paths between two nodes in a directed graph using DFS
+ * @details
+ * This algorithm implements Depth First Search (DFS) to count the number of 
+ * possible paths between two nodes in a directed graph. It is represented using 
+ * an adjacency matrix. The algorithm recursively traverses the graph to find 
+ * all paths from the source node `u` to the destination node `v`.
+ * 
+ * @author [Aditya Borate](https://github.com/adi776borate)
+ * @see https://en.wikipedia.org/wiki/Path_(graph_theory)
+ */
+
+#include <vector>   /// for std::vector
+#include <iostream> /// for IO operations
+#include <cassert>  /// for assert
+
+/**
+ * @namespace graph
+ * @brief Graph algorithms
+ */
+namespace graph {
+
+    /**
+     * @brief Helper function to perform DFS and count the number of paths from node `u` to node `v`
+     * @param A adjacency matrix representing the graph (1: edge exists, 0: no edge)
+     * @param u the starting node
+     * @param v the destination node
+     * @param n the number of nodes in the graph
+     * @param visited a vector to keep track of visited nodes in the current DFS path
+     * @returns the number of paths from node `u` to node `v`
+     */
+    int count_paths_dfs(const std::vector<std::vector<unsigned int>>& A, int u, int v, int n, std::vector<bool>& visited) {
```

Comment:
```suggestion
    int count_paths_dfs(const std::vector<std::vector<unsigned int>>& A, int u, int v, int n, std::vector<bool>& visited) {
```
you've used `unsigned int` but thats a variable sized integer, Prefer fixed size integer
for example
```
std::uint32_t number // equivalent to `unsigned int number`

Reply:
I have made the changes as you requested; please review them. Thank you.